src/gpu/multiexp.rs

```diff
@@ -224,11 +224,11 @@ where
     pub fn create(priority: bool) -> GPUResult<MultiexpKernel<E>> {
         let lock = locks::GPULock::lock();
 
-        let devices = opencl::Device::all()?;
+        let devices = opencl::Device::all();
 
         let kernels: Vec<_> = devices
             .into_iter()
-            .map(|d| (d.clone(), SingleMultiexpKernel::<E>::create(d, priority)))
+            .map(|d| (d, SingleMultiexpKernel::<E>::create(d.clone(), priority)))
```

Comment:
This strikes me as a less correct ordering.  What's going on here?

Reply:
The `devices` are now references. for getting the device name for the error, a reference is fine. The `SingleMultiExpKernel` needs an owned value, hence the clone.